### PR TITLE
Add STP bpduguard to port channels and interfaces

### LINKED ISSUE #716: Support of spanning-tree bpduguard in eos_l3ls_evpn port profiles in SERVERS.yml
**Issue Type**

- [x] Template enhancement

**Describe the solution you'd like**
Support of spanning-tree bpduguard enabled to fabric edge ports to enable a kind of loop detection.
This should be available via port-profiles and/or individual server definitions in the DC_SERVERS.yml file.
Feature should be available on individual ethernet interfaces and port-channels.

spanning_tree_bpduguard: < true | false > 
should render "spanning-tree bpduguard enabled"

port_profiles:
  < port_profile_1 >:
    "snip"
    spanning_tree_portfast: < edge | network >
    spanning_tree_bpdufilter: < true | false >
    **spanning_tree_bpduguard: < true | false >** 

**Describe alternatives you've considered**
I might enable it globally for all interfaces via the eos_cli_config_gen template
spanning_tree:
  edge_port:
    bpduguard_default: < true | false >

and disable it manually individually on (edge) ports, to which addtional switches are connected to.

### PR DESCRIPTION
**Pull request by melkypie**
## Change Summary
Adds `spanning-tree bpduguard true/false` to port-channels and server interfaces

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

Fixes #716

## Component(s) name

`arista.avd.eos_designs`

## Checklist:

- [X] My code has been rebased from devel before I start
- [X] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [X] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

File: ansible_collections/arista/avd/molecule/evpn_underlay_ebgp_overlay_ebgp/intended/structured_configs/DC1-LEAF1A.yml [hunk 1]
```diff
@@ -177,6 +177,7 @@ ethernet_interfaces:
     vlans: 1-4094
     spanning_tree_portfast: edge
     spanning_tree_bpdufilter: true
+    spanning_tree_bpduguard: false
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/evpn_underlay_ebgp_overlay_ebgp/intended/structured_configs/DC1-LEAF1A.yml:180]: I don't see an update of the group_var files in this molecule testcase, so how are these changes implemented? The structured_configs are part of the output of the molecule, so they should not be changed directly.
You should change the "DC1_SERVERS.yml" group_var file to include this new knob, and then run the molecule scenarios using "make refresh-facts" from the molecule main folder.

Comment (melkypie) [reply ansible_collections/arista/avd/molecule/evpn_underlay_ebgp_overlay_ebgp/intended/structured_configs/DC1-LEAF1A.yml:180]: I noticed that too, but if you look in the DC1_SERVERS.yml you can see that someone had spanning_tree_bpduguard added before this to multiple parts.

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/evpn_underlay_ebgp_overlay_ebgp/intended/structured_configs/DC1-LEAF1A.yml:180]: Haha good point. Then please just run the "make refresh-facts" so all molecule test cases run with your changes, and commit the resulting changes. Then you should be good.
Other than that, everything looks good to me!

**Review by ClausHolbechArista — APPROVED** [review]
LGTM! Thank you for this contribution!

**Comment by ClausHolbechArista** [comment]
Suggestion for improvement. If `spanning_tree_bpduguard == false` we should specifically disable bpduguard on the interface. This would be useful in cases where global knob has been configured to enable bpduguard on all interfaces with portfast, but you wish to disable bpdugaurd explicitly on one interface.
Also FYI we will hold back most PRs until 2.0.0 has been released.

**Comment by melkypie** [comment]
> Suggestion for improvement. If `spanning_tree_bpduguard == false` we should specifically disable bpduguard on the interface. This would be useful in cases where global knob has been configured to enable bpduguard on all interfaces with portfast, but you wish to disable bpdugaurd explicitly on one interface.
> Also FYI we will hold back most PRs until 2.0.0 has been released.

Am I correct in assuming that this would be a change in eos_cli_config_gen? Maybe best to open another PR with an elif added to `spanning-tree bpduguard` and `spanning-tree bpdufilter` for them be set to `disable` on `false`
I am fine with PRs being postponed to v2.1.0 as that will allow us to start using v2.0.0 in our setup.

**Pull request merged**